### ISSUE TITLE
Why always yolo11n.pt?

### ISSUE THREAD

**Issue by XCJ-LateFall** [ultralytics/ultralytics#20946]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I trained different models and set pretrained=True for all of them during the training process, but pre-trained weights downloaded were all yolo11n.pt. Shouldn't different models download different pre-training weights?

### Additional

The following is my train.py
from ultralytics import YOLO,RTDETR
import os
import torch
os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'

if __name__ == '__main__':
    model_yaml = 'ultralytics/cfg/models/rt-detr/rtdetr-l.yaml'
    # model_yaml = 'ultralytics/cfg/models/12/yolo12.yaml'  # √
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_CBAM.yaml'  # √
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_EMA.yaml'
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_SE.yaml'
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_ECA.yaml'
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_CA.yaml'  # √
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_SimAM.yaml'  # √
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_S2Attention.yaml'
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_NAM.yaml'  # √
    # model_yaml = 'ultralytics/cfg/models/mycfg/yolov12_CrissCrossAttention.yaml'  # √

    # model = YOLO(model_yaml)
    model = RTDETR(model_yaml)  
    model.info()

    model.train(data='data.yaml',
                epochs=300,
                batch=8,
                imgsz=640,
                # optimizer='AdamW',
                pretrained=True,
                )

**Comment by UltralyticsAssistant** [comment]
👋 Hello @XCJ-LateFall, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will assist you here soon to help with your question about pretrained weights and model selection!

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/17011

**Comment by XCJ-LateFall** [comment]
> Duplicate of [#17011](https://github.com/ultralytics/ultralytics/issues/17011)

Thank you! 
Now I know that yolo11n.pt is used for AMP Check. However, when I set pretrained=True, the pretrained weights corresponding to the model are not automatically downloaded. If I want to use the pretrained weights during training, how should I set the pretrained parameters?

**Comment by Y-T-G** [comment]
If you use `.pt`, it will use pretrained weights. If you use `.yaml`, it will train from scratch.

**Comment by XCJ-LateFall** [comment]
> If you use `.pt`, it will use pretrained weights. If you use `.yaml`, it will train from scratch.
Thank you for your reply！
So, what is the role of the pretrained parameters? What will happen if I use `.yaml` and set `pretrained=True` ?

**Comment by Y-T-G** [comment]
If you set it to `False` with `.pt`, it will reset the weights and train from scratch. But setting it to `True` doesn't have any effect. It's `True` by default. You can also provide path to an existing weights to `pretrained` and it will load the state dict or matching weights from that model.
